hyperspy/drawing/widgets.py

```diff
@@ -525,11 +527,14 @@ def update_range(self):
                       self.rect.get_x() + self.rect.get_width())
 
     def move_left(self, event):
-        if self.buttonDown is False or self.ignore(event):
+        if self.buttonDown is False or self.ignore(event) or\
+                event.xdata is None:
             return
         # Do not move the left edge beyond the right one.
         if event.xdata >= self.range[1]:
             return
+        if self.left and event.xdata < self.left:
```

Comment:
This need to test explicitly for `self.left is None`, or it will fail if its 0. Same for right etc. below.